src/redux/api/courtsApi.tsx

```diff
@@ -0,0 +1,38 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+import { environment } from "../../constants/environment";
+import { ICourt } from "../../interfaces/courtData";
+
+export const courtsApi = createApi({
+  reducerPath: "courtData",
+  baseQuery: fetchBaseQuery({ baseUrl: environment.apiBaseUrl }),
+  tagTypes: ["courts"],
+  endpoints: (builder) => ({
+    getAllCourt: builder.query({
+      query: () => "/courts",
+      providesTags: [{ type: "courts", id: "LIST" }],
+    }),
+    updateCourt: builder.mutation({
+      query: (court: ICourt) => ({
+        url: `/courts/${court._id}`,
+        method: "PUT",
+        body: court,
+      }),
+      invalidatesTags: (_1, _2, court) => [
+        { type: "courts", id: court._id },
```

Comment:
I don't think it is necessary to have this invalid tag, as you only defined [{ type: "courts", id: "LIST" }] in the builder.query()

Reply:
The[type:"courts", id:court._id] here is used to deactivate the related court with this id only, it will not affect the courts' cache with other ids to improve the performance. for example, if the court with id "1" was updated by an admin and the cache of that court only will be updated, and the cache of the court with id "2" will not be affected.If the [type: "courts", id:court._id] is not defined in this invalid tag, when the court with id "1" was updated, not only the cache of the court with id "1" be removed, but also the court with id "2" be affected.  This is why [type:"courts", id:court._id] was defined here, especially when getCourtById should be used later.

Reply:
> court

👍

Reply:
> The[type:"courts", id:court._id] here is used to deactivate the related court with this id only, it will not affect the courts' cache with other ids to improve the performance. for example, if the court with id "1" was updated by an admin and the cache of that court only will be updated, and the cache of the court with id "2" will not be affected.If the [type: "courts", id:court._id] is not defined in this invalid tag, when the court with id "1" was updated, not only the cache of the court with id "1" be removed, but also the court with id "2" be affected. This is why [type:"courts", id:court._id] was defined here, especially when getCourtById should be used later.

👍